chainer/optimizers/adam.py

```diff
@@ -164,7 +194,45 @@ def update_core_gpu(self, param):
             raise ValueError(
                 'eps of Adam optimizer is too small for {} ({})'.format(
                     grad.dtype.name, hp.eps))
-        if hp.amsgrad:
+        if hp.adabound:
+            lower, upper = self.bounds
+        if hp.amsgrad and hp.adabound:
+            if AdamRule._amsbound_kernel is None:
+                AdamRule._amsbound_kernel = cuda.elementwise(
+                    'T grad, T alpha_t, T one_minus_beta1, T one_minus_beta2, '
+                    'T lower, T upper, '
+                    'T eps, T eta, T weight_decay_rate',
+                    'T param, T m, T v, T vhat',
+                    '''m += one_minus_beta1 * (grad - m);
+                       v += one_minus_beta2 * (grad * grad - v);
+                       vhat = max(vhat, v);
+                       param -= eta * (m * max(lower, min(upper, alpha_t / (sqrt(vhat) + eps))) +
```

Comment:
Could you refactor this line to pass Python static check?
I guess one reference is L244-245 of this file.

> /home/travis/build/chainer/chainer/chainer/optimizers/adam.py:209:80: E501 line too long (97 > 79 characters)